src/network/NetworkKey.cpp

```diff
@@ -0,0 +1,436 @@
+        return false;
+    }
+    int result = PEM_write_bio_RSAPrivateKey(bio, rsa, NULL, NULL, 0, NULL, NULL);
+    if (result != 1)
+    {
+        log_error("failed to write private key!");
+        BIO_free_all(bio);
+        return false;
+    }
+    RSA_free(rsa);
+
+    int keylen = BIO_pending(bio);
+    char * pem_key = new char[keylen];
+    BIO_read(bio, pem_key, keylen);
+    file->write(file, pem_key, keylen, 1);
+    log_verbose("saving key of length %u", keylen);
+    BIO_free_all(bio);
+    delete [] pem_key;
+#else
+    #error Only RSA is supported!
+#endif
+
+    return true;
+}
+
+bool NetworkKey::SavePublic(SDL_RWops *file)
+{
+    if (m_key == nullptr)
+    {
+        log_error("No key loaded");
+        return false;
+    }
+    RSA * rsa = EVP_PKEY_get1_RSA(m_key);
+    if (rsa == nullptr)
+    {
+        log_error("Failed to get RSA key handle!");
+        return false;
+    }
+    BIO * bio = BIO_new(BIO_s_mem());
+    if (bio == nullptr)
+    {
+        log_error("Failed to initialise OpenSSL's BIO!");
+        return false;
+    }
+    int result = PEM_write_bio_RSAPublicKey(bio, rsa);
+    if (result != 1)
+    {
+        log_error("failed to write private key!");
+        BIO_free_all(bio);
+        return false;
+    }
+    RSA_free(rsa);
+
+    int keylen = BIO_pending(bio);
+    char * pem_key = new char[keylen];
+    BIO_read(bio, pem_key, keylen);
+    file->write(file, pem_key, keylen, 1);
+    BIO_free_all(bio);
+    delete [] pem_key;
+
+    return true;
+}
+
+std::string NetworkKey::PublicKeyString()
+{
+    if (m_key == nullptr)
+    {
+        log_error("No key loaded");
+        return nullptr;
+    }
+    RSA * rsa = EVP_PKEY_get1_RSA(m_key);
+    if (rsa == nullptr) {
+        log_error("Failed to get RSA key handle!");
+        return nullptr;
+    }
+    BIO * bio = BIO_new(BIO_s_mem());
+    if (bio == nullptr)
+    {
+        log_error("Failed to initialise OpenSSL's BIO!");
+        return nullptr;
+    }
+    int result = PEM_write_bio_RSAPublicKey(bio, rsa);
+    if (result != 1)
+    {
+        log_error("failed to write private key!");
+        BIO_free_all(bio);
+        return nullptr;
+    }
+    RSA_free(rsa);
+
+    int keylen = BIO_pending(bio);
+    char * pem_key = new char[keylen + 1];
+    BIO_read(bio, pem_key, keylen);
+    BIO_free_all(bio);
+    pem_key[keylen] = '\0';
+    std::string pem_key_out(pem_key);
+    delete [] pem_key;
+
+    return pem_key_out;
+}
+
+/**
+ * @brief NetworkKey::PublicKeyHash
+ * Computes a short, human-readable (e.g. asciif-ied hex) hash for a given
+ * public key. Serves a purpose of easy identification keys in multiplayer
+ * overview, multiplayer settings.
+ *
+ * In particular, any of digest functions applied to a standarised key
+ * representation, like PEM, will be sufficient.
+ *
+ * @return returns a string containing key hash.
+ */
+std::string NetworkKey::PublicKeyHash()
+{
+    std::string key = PublicKeyString();
+    if (key.empty()) {
```

Comment:
How about a proper tool for that instead of having to do such menial checks and changes manually? I think there's already an issue for that.